### ISSUE TITLE
Surgery with derivingVia

### ISSUE THREAD

**Issue by Drezil** [Lysxia/generic-data#13]
We can derive many things using derivingVia and this package.

Is it possible to derive surgically altered things with deriving via? I could not get it compiling..

i.e. instead of
```haskell
data T a = T { unT :: a }

instance (Show a) => Show (T a) where
  showsPrec n = gshowsPrec n . derecordify . toData
```
something like
```haskell
data T a = T { unT :: a }
  deriving Show via Derecordify T a
```
latter yields the error
```
error:
    • Couldn't match representation of type ‘T a’
                               with that of ‘Derecordify T a’
        arising from the coercion of the method ‘showsPrec’
          from type ‘Int -> Derecordify T a -> ShowS’
            to type ‘Int -> T a -> ShowS’
    • When deriving the instance for (Show (T a))
    |
xxx |   deriving Show via Derecordify T a
    |            ^^^^
```

Is it just a missing instance `Coercible (Derecordify a) a` and `Coercible a (Derecordify a)`? Or is this harder to fix?

**Comment by Lysxia** [comment]
It's a bit more involved, we need to add a newtype for this, but nothing too challenging. It would look like

    data T a = T { unT :: a }
      deriving Show via Surgery Derecordify (T a)

for some newtype `Surgery` and after making `Derecordify` a defunctionalized symbol (to be able to reuse its name).

**Comment by Lysxia** [comment]
How's #14 look?

**Comment by Drezil** [comment]
Cool! Thank you!

**Comment by Lysxia** [comment]
I just released `0.6.0.0`!
